src/components/header-text-with-dropdown/HeaderTextWithDropdown.tsx

```diff
@@ -0,0 +1,115 @@
+import { Dispatch, FocusEvent, SetStateAction, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import MenuItem from '@mui/material/MenuItem'
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp'
+import IconButton from '@mui/material/IconButton'
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+import { menuTypes } from '~/containers/course-section/CourseSectionContainer.constants'
+import useMenu from '~/hooks/use-menu'
+
+import {
+  TextFieldVariantEnum,
+  ColorEnum,
+  CourseSection,
+  FormInputValueChange
+} from '~/types'
+import { styles } from '~/components/header-text-with-dropdown/HeaderTextWithDropdown.styles'
+
+interface HeaderTextWithDropdownProps {
+  sectionData: CourseSection
+  setIsVisible: Dispatch<SetStateAction<boolean>>
+  onDeleteSection?: () => void
+  handleSectionInputChange?: FormInputValueChange<string, CourseSection>
+  isVisible: boolean
+  isView?: boolean
+}
+
+const HeaderTextWithDropdown = ({
+  setIsVisible,
+  handleSectionInputChange,
+  sectionData,
+  onDeleteSection,
+  isVisible,
+  isView = false
+}: HeaderTextWithDropdownProps) => {
+  const { t } = useTranslation()
+  const [titleInput, setTitleInput] = useState(sectionData.title)
+  const [activeMenu, setActiveMenu] = useState('')
+  const { openMenu, renderMenu } = useMenu()
+
+  const onShowHide = () => {
+    setIsVisible((isVisible) => !isVisible)
+  }
+
+  const handleBlur = (
+    event: FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    if (handleSectionInputChange && !isView)
+      handleSectionInputChange(sectionData.id, 'title', event.target.value)
+  }
+
+  const sectionActions = [
+    {
+      id: 1,
+      label: (
+        <Box sx={styles.deleteIconWrapper}>
+          <DeleteOutlineIcon sx={styles.menuIcon} />
+          {t('course.courseSection.sectionMenu.deleteSection')}
+        </Box>
+      ),
+      onClick: onDeleteSection
+    }
+  ]
+
+  const sectionMenuItems = sectionActions.map(({ label, onClick, id }) => (
+    <MenuItem key={id} onClick={onClick}>
+      {label}
+    </MenuItem>
+  ))
+
+  const showIcon = isVisible ? (
+    <KeyboardArrowUpIcon color={ColorEnum.Primary} sx={styles.headerIcon} />
+  ) : (
+    <KeyboardArrowDownIcon color={ColorEnum.Primary} sx={styles.headerIcon} />
+  )
+  const showMenu = isView
```

Comment:
`showIcon` for me sounds like an action and not like a true \ false flag. `isIconVisible` or smth will be better in such cases